metals/src/main/scala/scala/meta/internal/metals/MetalsSymbolSearch.scala

```diff
@@ -132,4 +132,61 @@ class MetalsSymbolSearch(
       Some(new BuildTargetIdentifier(buildTargetIdentifier)),
     )
   }
+
+  override def queryImplicitClassMembers(
+      paramTypeSymbol: String
+  ): ju.List[scala.meta.pc.ImplicitClassMemberResult] = {
+    import scala.meta.internal.jdk.CollectionConverters._
+
+    scribe.info(
+      s"[MetalsSymbolSearch] Querying implicit class members for type: $paramTypeSymbol"
+    )
+
+    val totalCached = wsp.implicitClassMembers.values.flatten.size
```

Comment:
I would do all that in the compiler itself, it's much easier to do there. I would just index any implicit classes and then check in the presentation compiler if they apply to the current member.